lerobot/common/robot_devices/cameras/ros2.py

```diff
@@ -0,0 +1,277 @@
+"""
+This file contains utilities for recording frames from cameras. For more info look at `ROS2Camera` docstring.
+"""
+
+import argparse
+import concurrent.futures
+import shutil
+import threading
+import time
+from pathlib import Path
+from threading import Thread
+
+# TODO(Yadunund): Implement mock.
+import cv2
+import numpy as np
+import rclpy
+from cv_bridge import CvBridge
+from PIL import Image
+from sensor_msgs.msg import Image as ImageMsg
+
+from lerobot.common.robot_devices.cameras.configs import ROS2CameraConfig
+from lerobot.common.robot_devices.utils import (
+    RobotDeviceAlreadyConnectedError,
+    RobotDeviceNotConnectedError,
+)
+from lerobot.common.utils.utils import capture_timestamp_utc
+
+
+def save_image(img_array, camera_name, frame_index, images_dir):
+    img = Image.fromarray(img_array)
+    path = images_dir / f"camera_{camera_name:02d}_frame_{frame_index:06d}.png"
+    path.parent.mkdir(parents=True, exist_ok=True)
+    img.save(str(path), quality=100)
+
+
+def save_images_from_cameras(
+    images_dir: Path,
+    topic: str | None = None,
```

Comment:
can we support multiple images here? 

The test code is located in here: https://github.com/huggingface/lerobot/blob/a75d00970f55b48061909e1b7fa825e27bfe8d06/tests/cameras/test_cameras.py#L196-L206